samples/extensions/shader_debugprintf/shader_debugprintf.cpp

```diff
@@ -393,9 +416,37 @@ bool ShaderDebugPrintf::prepare(const vkb::ApplicationOptions &options)
 	return true;
 }
 
+// This sample overrides the per-sample layer framework to force activation of the validation layer
```

Comment:
Why not always simply return VK_LAYER_KHRONOS_validation?

Reply:
I think, for release builds, you usually don't want the validation layer to be active.

Reply:
I did this so that `VK_LAYER_KHRONOS_validation` is not enabled twice when in Debug mode and the log does not show the same validation layer twice in a row.  Otherwise you get this:

```
[info] Initializing Vulkan sample
[info] Vulkan debug utils enabled (VK_EXT_debug_utils)
[info] Extension VK_EXT_debug_utils found, enabling it
[info] Extension VK_KHR_get_physical_device_properties2 found, enabling it
[info] Extension VK_KHR_portability_enumeration found, enabling it
[info] Extension VK_EXT_metal_surface found, enabling it
[info] Extension VK_EXT_layer_settings found, enabling it
[info] Enabled Validation Layers:
[info] 		VK_LAYER_KHRONOS_validation
[info] 		VK_LAYER_KHRONOS_validation
```